src/_pytest/pytester.py

```diff
@@ -1089,14 +1096,42 @@ def collect_by_name(
                 return colitem
         return None
 
+    @overload
     def popen(
         self,
         cmdargs,
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE,
-        stdin=CLOSE_STDIN,
+        stdout: Optional[Union[int, IO]] = subprocess.PIPE,
```

Comment:
Is it specifically `IO[str]` or `IO[bytes]`?

I see that typeshed punts on this: https://github.com/python/typeshed/blob/d8b081130d3e3a7fde1a49ac787d304f5c4c1966/stdlib/3/subprocess.pyi#L13-L27

Reply:
I've used `AnyStr` here initially (but forgot to clean it up).
I've figured `IO` is the same as `IO[Any]`.  Do you prefer the latter?
We basically have the `_FILE` from typeshed here: `Union[None, int, IO[Any]]`

Reply:
I'd leave it with the explicit `Any` (like you did), maybe can be filled with something precise in the future.

Reply:
It can be `str` or `bytes`.  typeshed binds it according to overloads.